kvbc/src/st_reconfiguration_sm.cpp

```diff
@@ -82,4 +88,17 @@ bool StReconfigurationHandler::handle(const concord::messages::AddRemoveWithWedg
   return true;
 }
 
+bool StReconfigurationHandler::handle(const concord::messages::PruneRequest &command, uint64_t bft_seq_num, uint64_t) {
+  // Actual pruning will be done from the lowest latestPruneableBlock returned by the replicas. It means, that even
+  // on every state transfer there might be at most one relevant pruning command. Hence it is enough to take the latest
+  // saved command and try to execute it
```

Comment:
This method will be called when ST completes and it will get the latest version of the key that contains the prune command. Then, it will prune up to the last agreed block ID in that command. Is my understanding correct?

Reply:
exactly